File: abc262/d.py
N = int(input())
a = list(map(int,input().split()))
MOD = 998244353
ans = 0
for i in range(1,N+1):
    dp = [[[0]*(i) for _ in range(N+1)] for _ in range(N+1)]
    dp[0][0][0] = 1
    for j in range(N):
        for k in range(j+1):
            for l in range(i):
                # j + 1項目を使う.
                dp[j+1][k][l] = (dp[j][k][l] + dp[j+1][k][l]) % MOD
                dp[j+1][k+1][(l+a[j])%i] = (dp[j][k][l] + dp[j+1][k+1][(l+a[j])%i]) % MOD

    ans = (ans + dp[N][i][0]) % MOD

print(ans)

File: abc262/d.cpp
#include <bits/stdc++.h>
#define ll long long
#define MOD 998244353
using namespace std;

int main()
{
    int N;
    cin >> N;
    vector<ll> a(N);
    for (int i = 0; i < N; i++)
    {
        cin >> a[i];
    }
    ll ans = 0;
    for (int i = 1; i <= N; i++)
    {
        vector dp(N + 1, vector(i + 1, vector<ll>(i, 0)));
        dp[0][0][0] = 1;
        for (int j = 0; j < N; j++)
        {
            for (int k = 0; k <= i; k++)
            {

                for (int l = 0; l < i; l++)
                {
                    dp[j + 1][k][l] = (dp[j][k][l] + dp[j + 1][k][l]) % MOD;
                    if (k < i)
                    {
                        dp[j + 1][k + 1][(l + a[j]) % i] = (dp[j][k][l] + dp[j + 1][k + 1][(l + a[j]) % i]) % MOD;
                    }
                }
            }
        }
        ans = (ans + dp[N][i][0]) % MOD;
    }
    cout << ans << endl;
}